docs/operations/setting_up_the_swarm.rst

```diff
@@ -202,4 +202,17 @@ Create Main Traefik Service
      --entrypoints='Name:http Address::80 Redirect.EntryPoint:https' \
      --entrypoints='Name:https Address::443 TLS:/etc/ssl/certs/openstax.crt,/etc/ssl/private/openstax.pem' \
      --logLevel=INFO \
-     --accessLog
\ No newline at end of file
+     --accessLog
+
+----
+
+***********************
+Set Up Required Secrets
+***********************
+
+The stack requires that the docker secret ``basic-auth-users`` is set in the swarm to work properly.
+An example of creating basic auth credentials with a single user is the following, when ``DOCKER_HOST`` is properly pointing to the running swarm:
+
+.. code-block:: bash
+
+  htpasswd -nbB <username> <password> | docker secret create basic-auth-users -
```

Comment:
Questions vs code comments: If we wanted to set multiple users over time or update passwords, would we basically rotate the secret? I don't see a way to retrieve the secret value (inspect seems to just give metadata 🤷‍♂️ ), so would we create a new password file with `htpasswd` and then push up? (e.g. in lieu of an append)

Reply:
That's correct. We would have to rotate (and that also means that everytime we do, we must know ALL existing credentials)

Reply:
Maybe we can keep a secret in aws secretsmanager, just to "remind us" what the existing secret value is when we need to rotate...